tls/s2n_client_hello.c

```diff
@@ -299,27 +318,52 @@ int s2n_process_client_hello(struct s2n_connection *conn)
     return S2N_SUCCESS;
 }
 
+static S2N_RESULT s2n_client_hello_process_cb_response(struct s2n_connection *conn, int rc)
+{
+    if (rc < 0) {
+        goto fail;
+    }
+    switch(conn->config->client_hello_cb_mode) {
+        case S2N_CLIENT_HELLO_CB_BLOCKING : {
+            if(rc) {
```

Comment:
we can, but that is potentially a breaking change. As we never checked if `rc==1` from blocking mode.

Reply:
I see. I agree, better to maintain backwards compatibility. In this case I think keeping the documentation specific is best then, even if our check is for any positive value.

Reply:
Shouldn't we update the documentation though? Based on the documentation, the customer will expect return 2 NOT to mark the server_hello used, but it will.

Reply:
updated the docs on the return code